### PR TITLE
Clean up some warnings from the test suite

### PR DESCRIPTION
**Pull request by Armavica**
**What is this PR about?**

This PR removes 200 warnings from the 678 generated by the test suite. This de-clutters the output of the tests and makes it easier to spot if something is wrong. Also, but I am not sure if this is significant, the whole test suite ran 10% faster on my computer after the cleanup. I will check if it also makes the github runners faster.

I think that the ones that I fixed are the less controversial ones.

A good proportion if not the majority of the remaining warnings comes from a numpy deprecation in aesara which I will also address.

The rest are less easy to fix, or potential legit indications of problems (when I was not sure, I left them alone).

If there is interest, I can make a second pass later and try to tackle the rest.

I can breakup this huge commit into smaller thematic ones, let me know if this would be easier to review.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Clean up some warnings generated by the test suite

### CODE REVIEW

**Comment by Armavica** [comment]
I am not sure why this test is failing on the runner, as it is passing on my computer.

**Comment by canyon289** [comment]
Thank you for the PR. Its because this one test is too sensitive. Feel free to double the relative tolerance to make it less flaky

image

**Comment by Armavica** [comment]
I see. Shouldn't that better be fixed by fixing the PRNG seed?

File: pymc/tests/test_idata_conversion.py [hunk 1]
```diff
@@ -320,7 +332,9 @@ def test_missing_data_model(self):
         model = pm.Model()
         with model:
             x = pm.Normal("x", 1, 1)
-            y = pm.Normal("y", x, 1, observed=data)
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", ImputationWarning)
```
Comment (ricardoV94) [inline pymc/tests/test_idata_conversion.py]: Instead of ignoring ImputationWarnings, could we confirm they are happening?

File: pymc/tests/test_quadpotential.py [hunk 2]
```diff
@@ -206,7 +209,9 @@ def test_full_adapt_sample_p(seed=4566):
     )
 
     n_samples = 1000
-    pot = quadpotential.QuadPotentialFullAdapt(2, np.zeros(2), m_inv, 1)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", ".*experimental feature.*", UserWarning)
```
Comment (ricardoV94) [inline pymc/tests/test_quadpotential.py]: I think we should also confirm this one is being issued

File: pymc/tests/test_sampling_jax.py [hunk 3]
```diff
@@ -16,17 +16,19 @@
...
-    _numpyro_nuts_defaults,
-    _replace_shared_variables,
-    _update_numpyro_nuts_kwargs,
-    get_jaxified_graph,
-    get_jaxified_logp,
-    sample_blackjax_nuts,
-    sample_numpyro_nuts,
-)
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", ".*module is experimental.*", UserWarning)
```
Comment (ricardoV94) [inline pymc/tests/test_sampling_jax.py]: Also should be confirmed

File: pymc/tests/test_variational_inference.py [hunk 4]
```diff
@@ -526,7 +526,9 @@ def init_(**kw):
 @pytest.fixture(scope="function")
 def inference(inference_spec, simple_model):
     with simple_model:
-        return inference_spec()
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", ".*experimental inference Operator.*", UserWarning)
```
Comment (ricardoV94) [inline pymc/tests/test_variational_inference.py]: Same here

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
This is amazing. There are just some cases where I think we should assert the warning is being issued instead of ignoring it.

**Comment by ricardoV94** [comment]
> I see. Shouldn't that better be fixed by fixing the PRNG seed?

I never saw this test fail so I think it's safe to just ignore and rerun.

**Comment by ricardoV94** [comment]
Opened an issue for the failing test: https://github.com/pymc-devs/pymc/issues/6069

**Comment by Armavica** [comment]
Thank you for your review! I pushed the requested changes.

**Review by ricardoV94 — APPROVED** [review]
